docs/tool-choice.md

```diff
@@ -0,0 +1,348 @@
+
+Forces the model to use a function tool. No output tools are sent.
+
+!!! warning "For direct model requests only"
+    Use `'required'` only with [direct model requests](direct.md), not with agent runs. Since output tools are excluded, an agent cannot complete normally.
+
+```python {test="skip"}
+from pydantic_ai import ModelRequest, ToolDefinition
+from pydantic_ai.direct import model_request_sync
+from pydantic_ai.models import ModelRequestParameters
+from pydantic_ai.settings import ModelSettings
+
+settings: ModelSettings = {'tool_choice': 'required'}
+params = ModelRequestParameters(
+    function_tools=[
+        ToolDefinition(
+            name='get_weather',
+            description='Get weather for a city',
+            parameters_json_schema={
+                'type': 'object',
+                'properties': {'city': {'type': 'string'}},
+                'required': ['city'],
+            },
+        )
+    ],
+    allow_text_output=True,
+)
+
+response = model_request_sync(
+    'openai:gpt-4o',
+    [ModelRequest.user_text_prompt('What is the weather in Paris?')],
+    model_settings=settings,
+    model_request_parameters=params,
+)
+```
+
+### `list[str]` - Specific tools
+
+Restricts the model to specific function tools by name. The model must use one of the listed tools. No output tools are sent.
+
+!!! warning "For direct model requests only"
+    Like `'required'`, use tool lists only with [direct model requests](direct.md).
+
+```python {test="skip"}
+from pydantic_ai import ModelRequest, ToolDefinition
+from pydantic_ai.direct import model_request_sync
+from pydantic_ai.models import ModelRequestParameters
+from pydantic_ai.settings import ModelSettings
+
+weather_tool = ToolDefinition(
+    name='get_weather',
+    description='Get weather for a city',
+    parameters_json_schema={'type': 'object', 'properties': {'city': {'type': 'string'}}},
+)
+time_tool = ToolDefinition(
+    name='get_time',
+    description='Get current time for a city',
+    parameters_json_schema={'type': 'object', 'properties': {'city': {'type': 'string'}}},
+)
+population_tool = ToolDefinition(
+    name='get_population',
+    description='Get population for a city',
+    parameters_json_schema={'type': 'object', 'properties': {'city': {'type': 'string'}}},
+)
+
+settings: ModelSettings = {'tool_choice': ['get_weather']}  # Only allow get_weather
+params = ModelRequestParameters(
+    function_tools=[weather_tool, time_tool, population_tool],
+    allow_text_output=True,
+)
+
+response = model_request_sync(
+    'openai:gpt-4o',
+    [ModelRequest.user_text_prompt('Tell me about Paris')],
+    model_settings=settings,
+    model_request_parameters=params,
+)
+# Model will use get_weather even though other tools are defined
+```
+
+### `ToolsPlusOutput` - Specific tools with output
+
+Restricts function tools while keeping output tools available. This is the recommended way to control tool choice in agent runs with structured output.
```

Comment:
I don't think we should call this "recommended", because it's unlikely to achieve what the user is really looking for. (I think the scenario where you want to require a tool call before output is much more common)

So yeah I think we should be more clear about the limitations and limited usefulness of the `tool_choice` setting, and explain that the rest requires agent-level features that are coming later. Deemphasizing it and not having a full doc for now makes sense to me; the full doc will be more appropriate for the agent-level feature.

- Also it's not just "with _structured_ output", right, but also with simple text output righ And images etc. So maybe just say "output"